File: easycontrol/app/src/main/java/top/saymzx/easycontrol/app/entity/AppData.java
package top.saymzx.easycontrol.app.entity;

import android.annotation.SuppressLint;
import android.content.ClipboardManager;
import android.content.Context;
import android.hardware.SensorManager;
import android.hardware.usb.UsbManager;
import android.net.wifi.WifiManager;
import android.os.Handler;
import android.view.WindowManager;

import top.saymzx.easycontrol.app.MainActivity;
import top.saymzx.easycontrol.app.adb.AdbKeyPair;
import top.saymzx.easycontrol.app.helper.DbHelper;
import top.saymzx.easycontrol.app.helper.PublicTools;

public class AppData {
  @SuppressLint("StaticFieldLeak")
  public static Context applicationContext;
  public static MainActivity mainActivity;
  public static Handler uiHandler;

  // 数据库工具库
  public static DbHelper dbHelper;

  // 密钥文件
  public static AdbKeyPair keyPair;

  // 系统服务
  public static ClipboardManager clipBoard;
  public static WifiManager wifiManager;
  public static UsbManager usbManager;
  public static WindowManager windowManager;
  public static SensorManager sensorManager;

  // 设置值
  public static Setting setting;

  public static void init(MainActivity m) {
    mainActivity = m;
    applicationContext = m.getApplicationContext();
    uiHandler = new android.os.Handler(m.getMainLooper());
    dbHelper = new DbHelper(applicationContext);
    clipBoard = (ClipboardManager) applicationContext.getSystemService(Context.CLIPBOARD_SERVICE);
    wifiManager = (WifiManager) applicationContext.getSystemService(Context.WIFI_SERVICE);
    usbManager = (UsbManager) applicationContext.getSystemService(Context.USB_SERVICE);
    windowManager = (WindowManager) applicationContext.getSystemService(Context.WINDOW_SERVICE);
    sensorManager = (SensorManager) applicationContext.getSystemService(Context.SENSOR_SERVICE);
    setting = new Setting(applicationContext.getSharedPreferences("setting", Context.MODE_PRIVATE));
    // 读取密钥
    keyPair = PublicTools.readAdbKeyPair();
  }

}

File: easycontrol/app/src/main/java/top/saymzx/easycontrol/app/entity/AppData.kt
package top.saymzx.easycontrol.app.entity

import android.app.Activity
import android.content.ClipboardManager
import android.content.Context
import androidx.room.Room
import kotlinx.coroutines.MainScope
import top.saymzx.easycontrol.adb.AdbKeyPair
import top.saymzx.easycontrol.app.BuildConfig
import top.saymzx.easycontrol.app.helper.NetHelper
import top.saymzx.easycontrol.app.helper.PublicTools
import top.saymzx.easycontrol.app.helper.SQLDatabase
import java.io.File

class AppData(var main: Activity) {

  // 全局协程域
  val mainScope = MainScope()

  // 公共工具库
  val publicTools = PublicTools(main)

  // 数据库工具
  val dbHelper = Room.databaseBuilder(main, SQLDatabase::class.java, "app").build()

  // 网络工具
  val netHelper = NetHelper()

  // 密钥文件
  var keyPair: AdbKeyPair
  val privateKey = File(main.applicationContext.filesDir, "private.key")
  val publicKey = File(main.applicationContext.filesDir, "public.key")

  // 剪切板
  val clipBoard = main.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager

  // 设置值
  var setting = Setting(main.getSharedPreferences("setting", Context.MODE_PRIVATE))

  // 当前版本号
  val versionCode = BuildConfig.VERSION_CODE

  init {
    // 读取密钥文件
    if (!privateKey.isFile || !publicKey.isFile) {
      AdbKeyPair.generate(privateKey, publicKey)
    }
    keyPair = AdbKeyPair.read(privateKey, publicKey)
  }

}